python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,285 @@
+
+        failed_actuators_id = []
+        failed_primary, failed_secondary = [], []
+        for i, actuator_id in enumerate(self.actuators_to_test):
+            await self.mtcs.assert_all_enabled()
+
+            # Determine if actuator has a secondary actuator
+            secondary_exist = self.has_secondary_actuator(actuator_id)
+
+            # Checkpoint
+            primary_index = self.m1m3_actuator_ids.index(actuator_id)
+            if secondary_exist:
+                secondary_index = self.m1m3_secondary_actuator_ids.index()
+                await self.checkpoint(
+                    f"Running bump test on DAA FA ID: {actuator_id} "
+                    f"(Primary/Secondary: {primary_index}/{secondary_index})"
+                )
+            else:
+                await self.checkpoint(
+                    f"Running bump test on SAA FA ID: {actuator_id} "
+                    f"(Primary {primary_index})"
+                )
+
+            try:
+                await self.mtcs.run_m1m3_actuator_bump_test(
+                    actuator_id=actuator_id,
+                    primary=True,
+                    secondary=secondary_exist,
+                )
+            # raise a RuntimeError if the bump test fails
+            except RuntimeError:
+                failed_actuators_id.append(actuator_id)
+                self.log.exception(f"Failed to run bump test on FA ID {actuator_id}.")
+
+            # Getting test status
+            (
+                fa_id,
+                primary_status,
+                secondary_status,
+            ) = await self.mtcs.get_m1m3_bump_test_status()
+
+            # Getting primary indexes for failures
+            if primary_status == BumpTest.FAILED:
+                failed_primary.append(actuator_id, primary_index)
+
+                self.log.info(
+                    f"Bump test done for {i + 1} of {len(self.actuators_to_test)}"
+                    f"Bump test on FA ID {fa_id}. Primary FA: "
+                    f"{primary_status.name.upper()}."
+                )
+
+            if secondary_exist:
+                if secondary_status == BumpTest.FAILED:
+                    failed_secondary.append(actuator_id, secondary_index)
+
+                    self.log.info(
+                        f"Bump test done for {i + 1} of {len(self.actuators_to_test)}"
+                        f"Bump test on FA ID {fa_id}. Primary/Secondary: "
+                        f"{primary_status.name.upper()}/{secondary_status.upper()} ."
+                    )
+
+        end_time = time.monotonic()
+        elapsed_time = (end_time - start_time) / 60.0
+
+        # Final checkpoint
+        await self.checkpoint(
+            f"M1M3 bump test completed. It took {elapsed_time:.2f} minutes"
+        )
+
+        # Final message with bump test results/status
+        if not failed_actuators_id:
+            self.log.info("All actuators PASSED the bump test.")
+        else:
+            failed_primary_str = " ".join(
```

Comment:
maybe better:
```
failed_primary_str = " ".join([f"{failed[0]}:{failed[1]} for failed in failed_primary"])
```
the use of str(item).replace is a bit non-standard, if Python changed str repr of tuple, it will ceased to work

Reply:
I think this is not going to work. Actually, the `failed_primary` list has the following format: `[(101, 1), (102, 2), (103, 3)]`. When we do the `failed_primary.append(actuator_id, index)`, we will get a failure, as the `append` method can only accept one argument, so we have to append tuples to the list, like this `failed_primary.append((actuator_id, index))`